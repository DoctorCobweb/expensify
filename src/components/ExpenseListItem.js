import React from 'react';
import { Link } from 'react-router-dom';

const ExpenseListItem = ({ description, amount, createdAt, id }) => {
  // console.log('props', dispatch);
  return (
    <li>
      <Link to={`/edit/${id}`}>
        <h3>{description}</h3>
      </Link>
      <p>{amount}</p>
      <p>{createdAt}</p>
    </li>
  );
};
export default ExpenseListItem;