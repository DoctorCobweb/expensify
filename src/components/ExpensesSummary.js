import React from 'react';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';

export const ExpensesSummary = (props) => {
  const count = props.expenseCount;
  const total = numeral(props.expensesTotal/100).format('$0,0.00');
  return (
    <div>
      { count === 1 ? 
          <h1>Viewing {count} expense totalling {total}</h1>
          :
          <h1>Viewing {count} expenses totalling {total}</h1>
      }
    </div>
  );
};

const mapStateToProps = (state) => {
  const selectedExpenses = selectExpenses(state.expenses, state.filters);
  return {
    expenseCount: selectedExpenses.length,
    expensesTotal: selectExpensesTotal(selectedExpenses)
  };
};


// get access to the state via props by calling 'connect' and passing in mapStateToProps
export default connect(mapStateToProps)(ExpensesSummary);