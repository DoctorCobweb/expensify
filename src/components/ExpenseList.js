import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';

const ExpenseList = (props) => (
  <div>
    <h1>expense list</h1>
    <ul>
      {
        props.expenses.map((expense) => <ExpenseListItem key={expense.id} {...expense} />)
      }
    </ul>
  </div>
);

const mapStateToProps = (state) => {
  console.log('state: ', state);
  return {
    expenses: selectExpenses(state.expenses, state.filters)
    // filters: state.filters
  };
};

export default connect(mapStateToProps)(ExpenseList);