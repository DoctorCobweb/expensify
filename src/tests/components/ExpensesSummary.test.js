import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should correclty display for one selected expense', () => {
  const wrapper = shallow(<ExpensesSummary expensesCount={1} expensesTotal={122}/>);
  expect(wrapper).toMatchSnapshot();
});

test('should correclty display for multiple-selected expenses', () => {
  const wrapper = shallow(<ExpensesSummary expensesCount={23} expensesTotal={1095100}/>);
  expect(wrapper).toMatchSnapshot();
});