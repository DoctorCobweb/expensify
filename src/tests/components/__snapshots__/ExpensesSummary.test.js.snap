// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correclty display for multiple-selected expenses 1`] = `
<div>
  <h1>
    Viewing 
     expenses totalling 
    $10,951.00
  </h1>
</div>
`;

exports[`should correclty display for one selected expense 1`] = `
<div>
  <h1>
    Viewing 
     expenses totalling 
    $1.22
  </h1>
</div>
`;
